magento/magento2
# Merge guest quote to customer quote 24808

### Description (*)

There is no possibility to merge guest cart and active customer cart in `\Magento\Quote\Model\QuoteManagement::assignCustomer()`. Method always throws error in case if customer has active cart. With this update carts will be merges and guest cart will be removed. This update includes updates for PHPUnit tests for assignCustomer method run and assignCustomer method run for non exists customer.
### Fixed Issues (if relevant)

1. magento/magento2#24808: How to Merge guest quote items to customer quote(if customer log in) by magento 2 rest API

### Manual testing scenarios (*)

1. Create a customer
2. Add any product to cart to get active cart or skip this step to get no active cart for customer (both cases covered)
3. Log out from customer account
4. Add any product to cart to get guest user cart
5. Get Authorization token for created user
6. Get guest user quote masked id and previously created customer entity id
7. Try to merge guest user cart with created customer cart using API request: 
8. Login with created customer again to check cart

### Questions or comments
`\Magento\Quote\Model\QuoteManagement::assignCustomer` is using `load` and `delete` methods from `\Magento\Quote\Model\QuoteIdMask` which Extends `AbstractModel` and implemented without service contract pattern. Maybe would be good to fix this with creation ResourceModel and required preferences, interfaces? 
### Contribution checklist (*)
